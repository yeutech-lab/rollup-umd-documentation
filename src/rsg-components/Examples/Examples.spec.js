import React from 'react';
import noop from 'lodash/noop';
import Examples from '../Examples';
import { DisplayModes } from '../../consts';

const examples = [
	{
		type: 'code',
		content: '<button>OK</button>',
		evalInContext: noop,
	},
	{
		type: 'markdown',
		content: 'Hello *world*!',
	},
];

it('should render examples', () => {
	const actual = shallow(<Examples examples={examples} name="button" />, {
		context: {
			codeRevision: 1,
			displayMode: DisplayModes.example,
		},
	});

	expect(actual).toMatchSnapshot();
});

it('should not render a example with unknown type', () => {
	const faultyExample = [
		{
			type: 'unknown',
			content: 'FooBar',
		},
	];

	const actual = mount(<Examples examples={faultyExample} name="button" />, {
		context: {
			codeRevision: 1,
		},
	});
	const article = actual.find('article');
	expect(article.length).toEqual(1);
	expect(article.text().includes('FooBar')).toEqual(false);
});
