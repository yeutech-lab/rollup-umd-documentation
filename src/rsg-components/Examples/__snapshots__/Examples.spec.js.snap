// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render examples 1`] = `
<Styled(Examples)>
  <Playground
    code=
<button>
  OK
</button>

    evalInContext={[Function]}
    index={0}
    key="1/0"
    name="button"
  />
  <Markdown
    key="1"
    text="Hello *world*!"
  />
</Styled(Examples)>
`;
